app/src/main/java/com/quran/labs/androidquran/model/quran/CoordinatesModel.java

```diff
@@ -141,7 +174,7 @@ private AyahCoordinates normalizePageAyahs(AyahCoordinates ayahCoordinates) {
       if (lastRect.left < firstRect.right) {
         RectF middleBounds = new RectF(lastRect.left,
             /* top= */ firstRect.bottom,
-            firstRect.right,
+            Math.min(firstRect.right, lastRect.right),
```

Comment:
instead of just taking `firstRect.right`, take the left most one. this fixes where the lines are drawn in very rare cases (like 1:7).